Fix native libgit2 library loading in libgit2sharp

On Ubuntu for Windows, a build with NB.GV fails with this error:

```
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load DLL 'git2-1196807': The specified module could not be found. [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:  (Exception from HRESULT: 0x8007007E) [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.LibraryLifetimeObject..ctor() [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    --- End of inner exception stack trace --- [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_repository_open(git_repository*& repository, FilePath path) [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.git_repository_open(String path) [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at LibGit2Sharp.Repository..ctor(String path, RepositoryOptions options, RepositoryRequiredParameter requiredParameter) [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.OpenGitRepo(String repoRoot) [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild) [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.Execute() [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/home/andrew/.nuget/packages/nerdbank.gitversioning/1.6.19-beta/build/Nerdbank.GitVersioning.targets(52,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() [/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj]
/usr/share/dotnet/sdk/1.0.1/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Sdk.targets(92,5): error : Cannot find project info for '/home/andrew/git/libgit2sharp/CodeGenerationAttributes/CodeGenerationAttributes.csproj'. This can indicate a missing project reference. [/home/andrew/git/libgit2sharp/LibGit2Sharp/LibGit2Sharp.csproj]
```

--- a/init.ps1
+++ b/init.ps1
@@ -1,6 +1,6 @@
 <#
 .SYNOPSIS
-Restores all NuGet, NPM and Typings packages necessary to build this repository. 
+Restores all NuGet, NPM and Typings packages necessary to build this repository.
 #>
 [CmdletBinding(SupportsShouldProcess)]
 Param(
@@ -18,12 +18,7 @@ try {
         & "$toolsPath\Restore-NuGetPackages.ps1" -Path $_ -Verbosity Quiet
     }
 
-    # Restore VS solution dependencies
-    gci "$PSScriptRoot\src" -rec |? { $_.FullName.EndsWith('.sln') } |% {
-        & "$toolsPath\Restore-NuGetPackages.ps1" -Path $_.FullName -Verbosity Quiet
-    }
-
-    # Restore VS2017 style as well, since nuget 3.3 doesn't support it.
+    # Restore VS2017 style to get the rest of the projects.
     msbuild "$PSScriptRoot\src\NerdBank.GitVersioning.Tests\NerdBank.GitVersioning.Tests.csproj" /t:restore /v:minimal /m /nologo
 
     Write-Host "Restoring NPM packages..." -ForegroundColor Yellow

--- a/src/MSBuildExtensionTask/ContextAwareTask.cs
+++ b/src/MSBuildExtensionTask/ContextAwareTask.cs
@@ -1,0 +1,111 @@
+﻿namespace MSBuildExtensionTask
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+#if NETCOREAPP1_0
+    using System.Runtime.Loader;
+#endif
+    using Microsoft.Build.Framework;
+    using Microsoft.Build.Utilities;
+
+    public abstract class ContextAwareTask : Task
+    {
+        protected virtual string ManagedDllDirectory => Path.GetDirectoryName(new Uri(this.GetType().GetTypeInfo().Assembly.CodeBase).LocalPath);
+
+        protected virtual string UnmanagedDllDirectory => null;
+
+        public override bool Execute()
+        {
+#if NETCOREAPP1_0
+            string taskAssemblyPath = new Uri(this.GetType().GetTypeInfo().Assembly.CodeBase).LocalPath;
+            var ctxt = new CustomAssemblyLoader(this);
+            Assembly inContextAssembly = ctxt.LoadFromAssemblyPath(taskAssemblyPath);
+            Type innerTaskType = inContextAssembly.GetType(this.GetType().FullName);
+            object innerTask = Activator.CreateInstance(innerTaskType);
+
+            var outerProperties = this.GetType().GetRuntimeProperties().ToDictionary(i => i.Name);
+            var innerProperties = innerTaskType.GetRuntimeProperties().ToDictionary(i => i.Name);
+            var propertiesDiscovery = from outerProperty in outerProperties.Values
+                                      where outerProperty.SetMethod != null && outerProperty.GetMethod != null
+                                      let innerProperty = innerProperties[outerProperty.Name]
+                                      select new { outerProperty, innerProperty };
+            var propertiesMap = propertiesDiscovery.ToArray();
+            var outputPropertiesMap = propertiesMap.Where(pair => pair.outerProperty.GetCustomAttribute<OutputAttribute>() != null).ToArray();
+
+            foreach (var propertyPair in propertiesMap)
+            {
+                object outerPropertyValue = propertyPair.outerProperty.GetValue(this);
+                propertyPair.innerProperty.SetValue(innerTask, outerPropertyValue);
+            }
+
+            var executeInnerMethod = innerTaskType.GetMethod(nameof(ExecuteInner), BindingFlags.Instance | BindingFlags.NonPublic);
+            bool result = (bool)executeInnerMethod.Invoke(innerTask, new object[0]);
+
+            foreach (var propertyPair in outputPropertiesMap)
+            {
+                propertyPair.outerProperty.SetValue(this, propertyPair.innerProperty.GetValue(innerTask));
+            }
+
+            return result;
+#else
+            // On .NET Framework (on Windows), we find native binaries by adding them to our PATH.
+            if (this.UnmanagedDllDirectory != null)
+            {
+                string pathEnvVar = Environment.GetEnvironmentVariable("PATH");
+                string[] searchPaths = pathEnvVar.Split(Path.PathSeparator);
+                if (!searchPaths.Contains(this.UnmanagedDllDirectory, StringComparer.OrdinalIgnoreCase))
+                {
+                    pathEnvVar += Path.PathSeparator + this.UnmanagedDllDirectory;
+                    Environment.SetEnvironmentVariable("PATH", pathEnvVar);
+                }
+            }
+
+            return this.ExecuteInner();
+#endif
+        }
+
+        protected abstract bool ExecuteInner();
+
+#if NETCOREAPP1_0
+        private class CustomAssemblyLoader : AssemblyLoadContext
+        {
+            private readonly ContextAwareTask loaderTask;
+
+            internal CustomAssemblyLoader(ContextAwareTask loaderTask)
+            {
+                this.loaderTask = loaderTask;
+            }
+
+            protected override Assembly Load(AssemblyName assemblyName)
+            {
+                string assemblyPath = Path.Combine(this.loaderTask.ManagedDllDirectory, assemblyName.Name) + ".dll";
+                if (File.Exists(assemblyPath))
+                {
+                    return LoadFromAssemblyPath(assemblyPath);
+                }
+
+                return Default.LoadFromAssemblyName(assemblyName);
+            }
+
+            protected override IntPtr LoadUnmanagedDll(string unmanagedDllName)
+            {
+                string unmanagedDllPath = Directory.EnumerateFiles(
+                    this.loaderTask.UnmanagedDllDirectory,
+                    $"{unmanagedDllName}.*").Concat(
+                        Directory.EnumerateFiles(
+                            this.loaderTask.UnmanagedDllDirectory,
+                            $"lib{unmanagedDllName}.*"))
+                    .FirstOrDefault();
+                if (unmanagedDllPath != null)
+                {
+                    return this.LoadUnmanagedDllFromPath(unmanagedDllPath);
+                }
+
+                return base.LoadUnmanagedDll(unmanagedDllName);
+            }
+        }
+#endif
+    }
+}

--- a/src/MSBuildExtensionTask/MSBuildExtensionTask.csproj
+++ b/src/MSBuildExtensionTask/MSBuildExtensionTask.csproj
@@ -1,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.0;net45</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Nerdbank.GitVersioning.LKG" Version="1.6.20-beta-gfea83a8c9e" PrivateAssets="all" />
+    <PackageReference Include="NuProj.Common" Version="0.11.14-beta" PrivateAssets="all" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net45' ">
+    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="14.3" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp1.0'">
+    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="15.1.548" />
+    <PackageReference Include="System.Runtime.Loader" Version="4.3.0" />
+  </ItemGroup>
+</Project>

--- a/src/NerdBank.GitVersioning/GitExtensions.cs
+++ b/src/NerdBank.GitVersioning/GitExtensions.cs
@@ -16,11 +16,6 @@
     public static class GitExtensions
     {
         /// <summary>
-        /// An AppDomain-wide variable used on 
-        /// </summary>
-        private static bool libgit2PathInitialized;
-
-        /// <summary>
         /// The 0.0 version.
         /// </summary>
         private static readonly Version Version0 = new Version(0, 0);
@@ -277,6 +272,7 @@
             return possibleCommits;
         }
 
+#if NET45
         /// <summary>
         /// Assists the operating system in finding the appropriate native libgit2 module.
         /// </summary>
@@ -308,28 +304,59 @@
         /// <returns><c>true</c> if the libgit2 native binaries have been found; <c>false</c> otherwise.</returns>
         public static bool TryHelpFindLibGit2NativeBinaries(string basePath, out string attemptedDirectory)
         {
-            attemptedDirectory = null;
-            if (!libgit2PathInitialized)
+            attemptedDirectory = FindLibGit2NativeBinaries(basePath);
+            if (Directory.Exists(attemptedDirectory))
             {
-#if !NET45
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-#endif
-                {
-                    attemptedDirectory = Path.Combine(basePath, "lib", "win32", IntPtr.Size == 4 ? "x86" : "x64");
-                    if (Directory.Exists(attemptedDirectory))
-                    {
-                        Environment.SetEnvironmentVariable("PATH", Environment.GetEnvironmentVariable("PATH") + Path.PathSeparator + attemptedDirectory);
-                    }
-                    else
-                    {
-                        return false;
-                    }
-                }
-
-                libgit2PathInitialized = true;
+                AddDirectoryToPath(attemptedDirectory);
+                return true;
             }
 
-            return libgit2PathInitialized;
+            return false;
+        }
+
+        /// <summary>
+        /// Add a directory to the PATH environment variable if it isn't already present.
+        /// </summary>
+        /// <param name="directory">The directory to be added.</param>
+        public static void AddDirectoryToPath(string directory)
+        {
+            Requires.NotNullOrEmpty(directory, nameof(directory));
+
+            string pathEnvVar = Environment.GetEnvironmentVariable("PATH");
+            string[] searchPaths = pathEnvVar.Split(Path.PathSeparator);
+            if (!searchPaths.Contains(directory, StringComparer.OrdinalIgnoreCase))
+            {
+                pathEnvVar += Path.PathSeparator + directory;
+                Environment.SetEnvironmentVariable("PATH", pathEnvVar);
+            }
+        }
+#endif
+
+        /// <summary>
+        /// Finds the directory that contains the appropriate native libgit2 module.
+        /// </summary>
+        /// <param name="basePath">The path to the directory that contains the lib folder.</param>
+        /// <returns>Receives the directory that native binaries are expected.</returns>
+        public static string FindLibGit2NativeBinaries(string basePath)
+        {
+#if !NET45
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+#endif
+            {
+                return Path.Combine(basePath, "lib", "win32", IntPtr.Size == 4 ? "x86" : "x64");
+            }
+#if !NET45
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return Path.Combine(basePath, "lib", "linux", IntPtr.Size == 4 ? "x86" : "x86_64");
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return Path.Combine(basePath, "lib", "osx");
+            }
+
+            return null;
+#endif
         }
 
         /// <summary>

--- a/src/NerdBank.GitVersioning/NerdBank.GitVersioning.csproj
+++ b/src/NerdBank.GitVersioning/NerdBank.GitVersioning.csproj
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
+    <TargetFrameworks>netcoreapp1.0;net45</TargetFrameworks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
   </PropertyGroup>
   <ItemGroup>

--- a/src/Nerdbank.GitVersioning.NuGet/Nerdbank.GitVersioning.NuGet.nuproj
+++ b/src/Nerdbank.GitVersioning.NuGet/Nerdbank.GitVersioning.NuGet.nuproj
@@ -87,8 +87,8 @@
         <TargetSubdirectory>MSBuildFull</TargetSubdirectory>
       </CrossTargetProjects>
       <CrossTargetProjects Include="@(_MSBuildProjectReferenceExistent)">
-        <SetPlatform>TargetFramework=netstandard1.3;%(_MSBuildProjectReferenceExistent.SetPlatform)</SetPlatform>
-        <SetTargetFramework>TargetFramework=netstandard1.3;ProjectHasSingleTargetFramework=false</SetTargetFramework>
+        <SetPlatform>TargetFramework=netcoreapp1.0;%(_MSBuildProjectReferenceExistent.SetPlatform)</SetPlatform>
+        <SetTargetFramework>TargetFramework=netcoreapp1.0;ProjectHasSingleTargetFramework=false</SetTargetFramework>
         <TargetSubdirectory>MSBuildCore</TargetSubdirectory>
       </CrossTargetProjects>
       <_MSBuildProjectReferenceExistent Remove="@(CrossTargetProjects)" />
@@ -97,7 +97,7 @@
     <!-- Build the default target of our P2P ref to ensure it builds. Hacky workaround.-->
     <MSBuild Projects="@(CrossTargetProjects)"
              BuildInParallel="true"
-             Properties="TargetFramework=netstandard1.3;ProjectHasSingleTargetFramework=false"
+             Properties="TargetFramework=netcoreapp1.0;ProjectHasSingleTargetFramework=false"
              Condition=" '$(BuildingInsideVisualStudio)' != 'true' " />
   </Target>
   <Target Name="SkipFrameworkAssemblies" AfterTargets="GetPackageFiles">
@@ -107,6 +107,7 @@
         !$([System.String]::new('%(TargetPath)').StartsWith('build\lib\')) and
         '%(FileName)' != 'NerdBank.GitVersioning' and
         '%(FileName)' != 'NerdBank.GitVersioning.Tasks' and
+        '%(FileName)' != 'MSBuildExtensionTask' and
         '%(FileName)' != 'LibGit2Sharp' and
         '%(FileName)' != 'Validation' and
         '%(FileName)' != 'Newtonsoft.Json' and

--- a/src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs
+++ b/src/Nerdbank.GitVersioning.Tasks/GetBuildVersion.cs
@@ -2,18 +2,13 @@
 {
     using System;
     using System.Collections.Generic;
-    using System.ComponentModel;
-    using System.Diagnostics;
     using System.IO;
     using System.Linq;
-    using System.Runtime.InteropServices;
-    using System.Text;
-    using System.Text.RegularExpressions;
     using Microsoft.Build.Framework;
     using Microsoft.Build.Utilities;
-    using Nerdbank.GitVersioning;
+    using MSBuildExtensionTask;
 
-    public class GetBuildVersion : Task
+    public class GetBuildVersion : ContextAwareTask
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="GetBuildVersion"/> class.
@@ -153,12 +148,12 @@
         [Output]
         public ITaskItem[] CloudBuildVersionVars { get; private set; }
 
-        public override bool Execute()
+        protected override string UnmanagedDllDirectory => GitExtensions.FindLibGit2NativeBinaries(this.TargetsPath);
+
+        protected override bool ExecuteInner()
         {
             try
             {
-                GitExtensions.TryHelpFindLibGit2NativeBinaries(this.TargetsPath);
-
                 var cloudBuild = CloudBuild.Active;
                 var oracle = VersionOracle.Create(Directory.GetCurrentDirectory(), this.GitRepoRoot, cloudBuild);
                 if (!string.IsNullOrEmpty(this.DefaultPublicRelease))
@@ -194,14 +189,14 @@
                         .Select(item => new TaskItem(item.Key, new Dictionary<string, string> { { "Value", item.Value } }))
                         .ToArray();
                 }
+
+                return !this.Log.HasLoggedErrors;
             }
             catch (ArgumentOutOfRangeException ex)
             {
                 Log.LogErrorFromException(ex);
                 return false;
             }
-
-            return true;
         }
     }
 }

--- a/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
+++ b/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
@@ -1,16 +1,17 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
+    <TargetFrameworks>netcoreapp1.0;net45</TargetFrameworks>
     <IsTool>true</IsTool>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="15.1.0-preview-000458-02" Condition=" '$(TargetFramework)' == 'netstandard1.3' " />
-    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="14.3" Condition=" '$(TargetFramework)' == 'net45' " />
-    <PackageReference Include="PInvoke.MSCorEE" Version="0.3.152" Condition=" '$(TargetFramework)' == 'net45' " />
     <PackageReference Include="Nerdbank.GitVersioning.LKG" Version="1.6.20-beta-gfea83a8c9e" />
     <PackageReference Include="NuProj.Common" Version="0.11.14-beta" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net45' ">
+    <PackageReference Include="PInvoke.MSCorEE" Version="0.3.152" />
+  </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\MSBuildExtensionTask\MSBuildExtensionTask.csproj" />
     <ProjectReference Include="..\NerdBank.GitVersioning\NerdBank.GitVersioning.csproj" />
   </ItemGroup>
 </Project>

--- a/src/Nerdbank.GitVersioning.sln
+++ b/src/Nerdbank.GitVersioning.sln
@@ -1,9 +1,9 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26215.1
+VisualStudioVersion = 15.0.26228.9
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FF286327-C783-4F7A-AB73-9BCBAD0D4460}") = "Nerdbank.GitVersioning.NuGet", "Nerdbank.GitVersioning.NuGet\Nerdbank.GitVersioning.NuGet.nuproj", "{B7DE4122-9E90-43DE-BE44-454315CFAE99}"
+Project("{5DD5E4FA-CB73-4610-85AB-557B54E96AA9}") = "Nerdbank.GitVersioning.NuGet", "Nerdbank.GitVersioning.NuGet\Nerdbank.GitVersioning.NuGet.nuproj", "{B7DE4122-9E90-43DE-BE44-454315CFAE99}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{4BD1A7CD-6F52-4F5A-825B-50E4D8C3ECFF}"
 	ProjectSection(SolutionItems) = preProject
@@ -23,6 +23,8 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NerdBank.GitVersioning", "NerdBank.GitVersioning\NerdBank.GitVersioning.csproj", "{C7FA7B7A-0469-4B1C-8657-E274C4CD8ABB}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Nerdbank.GitVersioning.Tasks", "Nerdbank.GitVersioning.Tasks\Nerdbank.GitVersioning.Tasks.csproj", "{B2454569-6EDC-4FD4-9936-D2B2F2E10409}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MSBuildExtensionTask", "MSBuildExtensionTask\MSBuildExtensionTask.csproj", "{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}"
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
@@ -84,6 +86,18 @@ Global
 		{B2454569-6EDC-4FD4-9936-D2B2F2E10409}.Release|x64.Build.0 = Release|Any CPU
 		{B2454569-6EDC-4FD4-9936-D2B2F2E10409}.Release|x86.ActiveCfg = Release|Any CPU
 		{B2454569-6EDC-4FD4-9936-D2B2F2E10409}.Release|x86.Build.0 = Release|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Debug|x64.ActiveCfg = Debug|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Debug|x64.Build.0 = Debug|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Debug|x86.Build.0 = Debug|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Release|Any CPU.Build.0 = Release|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Release|x64.ActiveCfg = Release|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Release|x64.Build.0 = Release|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Release|x86.ActiveCfg = Release|Any CPU
+		{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}.Release|x86.Build.0 = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE

